vault/src/main/java/org/opencadc/vault/metadata/DataNodeSizeSync.java

```diff
@@ -0,0 +1,225 @@
+                    stateDAO.put(state);
+                    state = stateDAO.get(state.getID());
+                    log.debug("created: " + state);
+                }
+                
+                long t1 = System.currentTimeMillis();
+                BackgroundLogInfo logInfo = new BackgroundLogInfo(instanceID.toString());
+                logInfo.setOperation(DataNodeSizeWorker.class.getSimpleName());
+                logInfo.setSuccess(false);
+                
+                // determine leader
+                boolean leader = checkLeaderStatus(state);
+                logInfo.leader = leader;
+                logInfo.setLastModified(state.curLastModified);
+                long sleep = LONG_SLEEP;
+                if (leader) {
+                    log.debug("leader: " + state);
+                    boolean fail = false;
+                    log.info(logInfo.start());
+                    try {
+                        DataNodeSizeWorker worker = new DataNodeSizeWorker(stateDAO, state, artifactDAO, artifactNamespace);
+                        worker.run();
+                        logInfo.setLastModified(state.curLastModified);
+                        logInfo.processed = worker.getNumArtifactsProcessed();
+                        logInfo.setSuccess(true);
+                    } catch (Exception ex) {
+                        log.error("unexpected worker fail", ex);
+                        fail = true;
+                    }
+                    
+                    leader = checkLeaderStatus(state);
+                    if (!fail && leader) {
+                        try {
+                            stateDAO.flushBufferedState();
+                        } catch (Exception ex) {
+                            log.error("unexpected HarvestState flush fail", ex);
+                            fail = true;
+                        }
+                    }
+                    
+                    if (!fail && leader) {
+                        try {
+                            // update leader timestamp before sleeping
+                            stateDAO.put(state, true);
+                        } catch (Exception ex) {
+                            log.error("unexpected HarvestState force update fail", ex);
+                            fail = true;
+                        }
+                    }
+
+                    if (fail) {
+                        sleep = FAIL_SLEEP;
+                    } else {
+                        sleep = SHORT_SLEEP;
+                    }
+                    logInfo.setSuccess(!fail);
+                    logInfo.setElapsedTime(System.currentTimeMillis() - t1);
+                } else {
+                    // not leader success
+                    sleep = LONG_SLEEP;
+                    logInfo.setSuccess(true);
+                }
+                logInfo.sleep = sleep;
+                log.info(logInfo.end());
+                Thread.sleep(sleep);
```

Comment:
Is there enough info in the logs to show the difference between the leader and the idle threads?

Reply:
The leader has a start and end log messages with `"leader":true`

The idle threads have only the end log message with  `"leader":false` (I chose that because the end also logs `"sleep":30000` (or longer for idle threads)